### DISCUSSION TITLE
let the user pick a folder then display the content and read it

### DISCUSSION THREAD

**Discussion by maddi115** [tauri-apps/tauri#9496] [Q&A] [answered]
can i pick a folder in my desktop full of mp3's and give it a interface with javascript
ive already tried setting true to the dialog config but i dont get how the syntax works should i just use bonjs for the job?
super new to tauri

**Comment by FabianLars** [comment]
For the dialog opening you're looking for this: https://tauri.app/v1/api/js/dialog#open (the second example, but with `multiple: false` is what you're asking for). This will give you the path to the folder.

Then for reading you can use https://tauri.app/v1/api/js/fs#readdir. How you display those paths/filenames is up to you though.

**Comment by maddi115** [comment]
1st error >>

      Uncaught TypeError: Failed to resolve module specifier "@tauri-apps/api/dialog". Relative references must start with either "/", "./", or "../".


2nd erorr>>

       Error: TypeError: appDir is not a function
    at getAppDir ((index):177:16)
    at HTMLButtonElement.<anonymous> ((index):197:37)

ive already installed yarn for the error still getting it tho

**Reply by FabianLars** [reply, marked as answer]
1st You can only consume npm packages if you use a bundler like vite or webpack. Without you'll have to change the import to something like this
```ts
const { open } = window.__TAURI__.dialog;
```

2nd - Are you on v2? If so, try appDataDir instead of appDir

**Reply by maddi115** [reply]
[✘] Environment
    - OS: Windows 10.0.19045 X64
    ✔ WebView2: 123.0.2420.97
    ✘ Couldn't detect any Visual Studio or VS Build Tools instance with MSVC and SDK components. Download from https://aka.ms/vs/17/release/vs_BuildTools.exe
    ✔ rustc: 1.77.2 (25ef9e3d8 2024-04-09)
    ✔ cargo: 1.77.2 (e52e36006 2024-03-26)
    ✔ rustup: 1.27.0 (bbb9276d2 2024-03-08)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 20.12.2
    - yarn: 1.22.22
    - npm: 10.5.0
    - bun: 1.1.4

[-] Packages
    - tauri [RUST]: 1.6.1
    - wry [RUST]: 0.24.7
    - tao [RUST]: 0.16.8
    - @tauri-apps/api : not installed!
    - @tauri-apps/cli [NPM]: 1.5.11

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../src
    - devPath: ../src

**Comment by maddi115** [comment]
yup that was it  and setting up configs in the tauri config json file thank you <333333333333  heres the github : https://github.com/anonq115/mp3-anom115
